subprojects/code-quality/src/integTest/groovy/org/gradle/api/plugins/quality/checkstyle/CheckstylePluginToolchainsIntegrationTest.groovy

```diff
@@ -0,0 +1,188 @@
+class CheckstylePluginToolchainsIntegrationTest extends AbstractIntegrationSpec {
+
+    def "uses jdk from toolchains"() {
+        given:
+        writeDummyConfig()
+        file('src/main/java/Dummy.java') << "class Dummy {}"
+        def jdk = setupExecutorForToolchains {
+            it.languageVersion > Jvm.current().javaVersion
+        }
+        writeBuildFile(jdk)
+
+        when:
+        succeeds("checkstyleMain")
+
+        then:
+        outputContains("Running checkstyle with toolchain '${jdk.javaHome.absolutePath}'.")
+    }
+
+    def "uses toolchains based on current jdk if not specified otherwise"() {
+        given:
+        writeDummyConfig()
+        writeBuildFile()
+
+        when:
+        succeeds("checkstyleMain")
+
+        then:
+        outputDoesNotContain("Running checkstyle with toolchain")
+    }
+
+    def "analyze good code"() {
+        goodCode()
+        def jdk = setupExecutorForToolchains {
+            it.languageVersion > Jvm.current().javaVersion
+        }
+        writeBuildFile(jdk)
+        writeConfigFileWithTypeName()
+
+        expect:
+        succeeds('checkstyleMain')
+        outputContains("Running checkstyle with toolchain '${jdk.javaHome.absolutePath}'.")
+        file("build/reports/checkstyle/main.xml").assertContents(containsClass("org.gradle.Class1"))
+        file("build/reports/checkstyle/main.xml").assertContents(containsClass("org.gradle.Class2"))
+
+        file("build/reports/checkstyle/main.html").assertContents(containsClass("org.gradle.Class1"))
+        file("build/reports/checkstyle/main.html").assertContents(containsClass("org.gradle.Class2"))
+    }
+
+    def "analyze bad code"() {
+        executer.withDefaultLocale(new Locale('en'))
+        badCode()
+        def jdk = setupExecutorForToolchains {
+            it.languageVersion > Jvm.current().javaVersion
+        }
+        writeBuildFile(jdk)
+        writeConfigFileWithTypeName()
+
+        expect:
+        fails("checkstyleMain")
+        outputContains("Running checkstyle with toolchain '${jdk.javaHome.absolutePath}'.")
+        failure.assertHasDescription("Execution failed for task ':checkstyleMain'.")
+        failure.assertHasErrorOutput("Name 'class1' must match pattern")
+        file("build/reports/checkstyle/main.xml").assertContents(containsClass("org.gradle.class1"))
+        file("build/reports/checkstyle/main.xml").assertContents(containsClass("org.gradle.class2"))
+
+        file("build/reports/checkstyle/main.html").assertContents(containsClass("org.gradle.class1"))
+        file("build/reports/checkstyle/main.html").assertContents(containsClass("org.gradle.class2"))
+    }
+
+    Jvm setupExecutorForToolchains(Spec<? super JvmInstallationMetadata> jvmFilter) {
+        Jvm jdk = AvailableJavaHomes.getAvailableJdk(jvmFilter)
+        Assume.assumeNotNull(jdk)
+        executer.beforeExecute {
+            withArguments("-Porg.gradle.java.installations.paths=${jdk.javaHome.absolutePath}", "--info")
```

Comment:
Is this necessary? If we disable auto-download, will this be able to find the JDK in the same way that AvailableJavaHomes does?

Reply:
It seems that for tests `-Porg.gradle.java.installations.paths` has to be set otherwise JDK is not found.